usr/src/uts/common/fs/zfs/dsl_dataset.c

```diff
@@ -1723,10 +1723,12 @@ get_clones_stat(dsl_dataset_t *ds, nvlist_t *nv)
 	zap_cursor_t zc;
 	zap_attribute_t za;
 	nvlist_t *propval = fnvlist_alloc();
-	nvlist_t *val = fnvlist_alloc();
+	nvlist_t *val;
 
 	ASSERT(dsl_pool_config_held(ds->ds_dir->dd_pool));
 
+	VERIFY0(nvlist_alloc(&val, 0, KM_SLEEP));
```

Comment:
Would you add a comment here explaining why we are not using fnvlist_alloc? Otherwise someone is likely to change it back later.

I think it would actually be fine to change fnvlist_alloc() to not use UNIQUE_NAME, but also understand if you don't want to do the testing for that potentially more impactful change right now.